Tools/run_batch.py

```
@@ -0,0 +1,314 @@
+                    subprocess.Popen(command),
+                )
+
+            finished_run_index = None
+            for run_index, (descriptor, process) in running_workers.items():
+                exit_code = process.poll()
+                if exit_code is None:
+                    continue
+
+                if exit_code != 0:
+                    raise RuntimeError(
+                        f"Run {run_index} exited with code {exit_code}.")
```

Ruff (0.15.12)

[warning] Abstract `raise` to an inner function

(TRY301)